analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpDiagnosticAnalyzerContextHelper.cs

```diff
@@ -26,84 +26,41 @@ namespace SonarAnalyzer.Helpers
-        }
 
-        public static void RegisterSyntaxTreeActionInNonGenerated(
-            this ParameterLoadingAnalysisContext context,
-            Action<SyntaxTreeAnalysisContext> action)
-        {
+        public static void RegisterSyntaxTreeActionInNonGenerated(this ParameterLoadingAnalysisContext context, Action<SyntaxTreeAnalysisContext> action) =>
             context.RegisterSyntaxTreeActionInNonGenerated(CSharpGeneratedCodeRecognizer.Instance, action);
-        }
 
-        public static void RegisterCodeBlockStartActionInNonGenerated<TLanguageKindEnum>(
-            this SonarAnalysisContext context,
-            Action<CodeBlockStartAnalysisContext<TLanguageKindEnum>> action) where TLanguageKindEnum : struct
-        {
+        public static void RegisterCodeBlockStartActionInNonGenerated<TSyntaxKind>(this SonarAnalysisContext context, Action<CodeBlockStartAnalysisContext<TSyntaxKind>> action)
+            where TSyntaxKind : struct =>
             context.RegisterCodeBlockStartActionInNonGenerated(CSharpGeneratedCodeRecognizer.Instance, action);
-        }
 
-        public static void ReportDiagnosticIfNonGenerated(
-            this CompilationAnalysisContext context,
-            Diagnostic diagnostic,
-            Compilation compilation)
-        {
+        public static void ReportDiagnosticIfNonGenerated(this CompilationAnalysisContext context, Diagnostic diagnostic, Compilation compilation) =>
             context.ReportDiagnosticIfNonGenerated(CSharpGeneratedCodeRecognizer.Instance, diagnostic, compilation);
-        }
 
-        public static void ReportDiagnosticIfNonGenerated(
-            this SymbolAnalysisContext context,
-            Diagnostic diagnostic,
-            Compilation compilation)
-        {
+        public static void ReportDiagnosticIfNonGenerated(this SymbolAnalysisContext context, Diagnostic diagnostic, Compilation compilation) =>
             context.ReportDiagnosticIfNonGenerated(CSharpGeneratedCodeRecognizer.Instance, diagnostic, compilation);
-        }
 
-        public static void ReportDiagnosticIfNonGenerated(
-            this SymbolAnalysisContext context,
-            Diagnostic diagnostic)
-        {
-            context.ReportDiagnosticIfNonGenerated(CSharpGeneratedCodeRecognizer.Instance, diagnostic,
-                context.Compilation);
-        }
+        public static void ReportDiagnosticIfNonGenerated(this SymbolAnalysisContext context, Diagnostic diagnostic) =>
+            context.ReportDiagnosticIfNonGenerated(CSharpGeneratedCodeRecognizer.Instance, diagnostic, context.Compilation);
 
-        internal static bool ShouldAnalyze(this SyntaxTree tree, AnalyzerOptions options,
-            Compilation compilation)
+        internal static bool ShouldAnalyze(this SyntaxTree tree, AnalyzerOptions options, Compilation compilation) =>
             // ToDo: PERF: do the global setting check before the per-file check.
```

Comment:
hm do we need a ticket for this?

Reply:
Trello card created

Reply:
There's a bunch of ToDo https://sonarcloud.io/project/issues?id=sonaranalyzer-dotnet&resolved=false&rules=csharpsquid%3AS1135